fabric-input-events-v1/src/main/java/net/fabricmc/fabric/impl/client/input/FabricKeyboardImpl.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.input;
+
+import net.minecraft.client.util.InputUtil;
+import net.minecraft.client.util.InputUtil.Key;
+
+public class FabricKeyboardImpl {
+	public static FabricKeyboardImpl INSTANCE = new FabricKeyboardImpl();
```

Comment:
Is this supposed to be accessed by end users? If so, make an interface that defines these methods, then have an instance that points to this.

Reply:
No, this is just the singleton implementation for `FabricKeyboard`. I guess it could be a static class instead, though.

Reply:
I don't think a singleton would be needed here since the client is also backed by a singleton. Static accessors via MinecraftClient.getInstance should be fine.